single_kernel_mongo/core/k8s_worload.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Kubernetes workload definition."""
+
+from itertools import chain
+from logging import getLogger
+from pathlib import Path
+
+from ops import Container
+from ops.pebble import ChangeError, ExecError
+from typing_extensions import override
+
+from single_kernel_mongo.config.literals import KubernetesUser
+from single_kernel_mongo.core.workload import WorkloadBase
+from single_kernel_mongo.exceptions import WorkloadExecError, WorkloadServiceError
+
+logger = getLogger(__name__)
+
+
+class KubernetesWorkload(WorkloadBase):
+    """Wrapper for performing common operations specific to the Kafka container."""
+
+    substrate = "k8s"
+    container: Container  # We always have a container in a Kubernetes Workload
```

Comment:
This only makes a type hint, not a ClassVar.
This is here because in the k8s case there should always be a Container while in the VM case there should never be a container so the smallest super type is `Container | None` and this provides the right type hints whenever it's needed.